### PR TITLE
Pairs don't flip to backside now

### PR DESCRIPTION
**Pull request by syss69**

### CODE REVIEW

File: src/App.test.tsx [hunk 1]
```diff
@@ -48,45 +48,74 @@ test('Card click callback', async() => {
...
-  const myImg = screen.getByRole("img", { name: /image_0/i });
+  const firstImagePair = screen.getAllByTestId("0");
+  const fipId =firstImagePair[0].getAttribute(`id`);
+  const fipId1 =  firstImagePair[1].getAttribute(`id`);
+  await userEvent.click(screen.getByRole("button", { name: /new game/i }))
+  const secondImagePair = screen.getAllByTestId("0");
+  const sipID = secondImagePair[0].getAttribute(`id`);
+  const sipID1 = secondImagePair[1].getAttribute(`id`);
+  expect(fipId).not.toEqual(sipID);
+  if(fipId === sipID){
```
Comment (clairecassan) [inline src/App.test.tsx:67]: This line won't be executed if the "expect" line 66 fails

File: src/App.test.tsx [hunk 2]
```diff
@@ -48,45 +48,74 @@ test('Card click callback', async() => {
...
+    expect(fipId1).not.toEqual(sipID1);
+  }
+  /*const myImg = screen.getByRole("img", { name: /^image_0$/i });
```
Comment (clairecassan) [inline src/App.test.tsx:70]: Always remove unnecessary code before submitting, unless you intend to use it later, or it serves a documentation purpose.

File: src/App.tsx [hunk 3]
```diff
@@ -1,37 +1,50 @@
 import React, {useState, useEffect} from "react";
 import "./App.css";
-import Card from "./Card";
+import Card, {Image} from "./Card";
+
+
+
+const images = [
+  {src:"https://api.dicebear.com/7.x/fun-emoji/svg?seed=Ginger", id:0},
```
Comment (clairecassan) [inline src/App.tsx:8]: as an exercise, create a const variable for "https://api.dicebar.com/7.x/fun-emoji/svg?see=", and use it to generate the 8 image names in the array

File: src/App.tsx [hunk 4]
```diff
@@ -1,37 +1,50 @@
...
+    images[3],
+    images[4],
+    images[5],
+    images[6],
+    images[7],
+  ]);
   const [callCount, setCallCount] = useState(0);
-  const [clicks, setClicks] = useState(1);
+  const [clicks, setClicks] = useState(1);  //clicks can't be more then 2, i use it to avoid click on 3rd card
+  const [clicks1, setClicks1] = useState(1);  // clicks1 used to count how many clicks do user for game
```
Comment (clairecassan) [inline src/App.tsx:43]: rename totalClickCount

File: src/App.tsx [hunk 5]
```diff
@@ -1,37 +1,50 @@
...
+    images[2],
+    images[3],
+    images[4],
+    images[5],
+    images[6],
+    images[7],
+  ]);
   const [callCount, setCallCount] = useState(0);
-  const [clicks, setClicks] = useState(1);
+  const [clicks, setClicks] = useState(1);  //clicks can't be more then 2, i use it to avoid click on 3rd card
```
Comment (clairecassan) [inline src/App.tsx:42]: rename: pairClickCount

File: src/App.tsx [hunk 6]
```diff
@@ -1,37 +1,50 @@
...
+    images[0],
+    images[1],
+    images[2],
+    images[3],
+    images[4],
+    images[5],
+    images[6],
+    images[7],
+  ]);
   const [callCount, setCallCount] = useState(0);
```
Comment (clairecassan) [inline src/App.tsx:41]: not clear what this is for

File: src/App.tsx [hunk 7]
```diff
@@ -97,11 +118,14 @@ function App() {
                 matchCount();
```
Comment (clairecassan) [inline src/App.tsx:118]: why wait for 500ms for that?

File: src/App.tsx [hunk 8]
```diff
@@ -70,7 +87,9 @@ function App() {
     }
   };
 
-
+  const setFalse = () => {
```
Comment (clairecassan) [inline src/App.tsx:90]: rename: hideAllCards

File: src/setupTests.ts [hunk 9]
```diff
@@ -3,3 +3,4 @@
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
 import '@testing-library/jest-dom';
+jest.setTimeout(10000);
```
Comment (clairecassan) [inline src/setupTests.ts:6]: I think we can remove that now. The new test implementation is faster

File: src/App.tsx [hunk 10]
```diff
@@ -38,9 +38,9 @@ function App() {
     images[6],
     images[7],
   ]);
-  const [callCount, setCallCount] = useState(0);
-  const [clicks, setClicks] = useState(1);  //clicks can't be more then 2, i use it to avoid click on 3rd card
-  const [clicks1, setClicks1] = useState(1);  // clicks1 used to count how many clicks do user for game
+  const [callCount, setCallCount] = useState(0); // callCount used for count matches. so, if matchCount called for 8 times - you win a game
```
Comment (clairecassan) [inline src/App.tsx:41]: so, shouldn't it be "matchCount"

File: src/App.tsx [hunk 11]
```diff
@@ -38,9 +38,9 @@ function App() {
...
+  const [pairClicks, setPairClicks] = useState(1);  //pairClicks can't be more then 2, i use it to avoid click on 3rd card
```
Comment (clairecassan) [inline src/App.tsx:42]: In code comments, do not use "I", prefer "We" or nothing, just write "to avoid click on 3rd card"

File: src/App.tsx [hunk 12]
```diff
@@ -1,18 +1,18 @@
...
 const images = [
-  {src:"https://api.dicebear.com/7.x/fun-emoji/svg?seed=Ginger", id:0},
-  {src:"https://api.dicebear.com/7.x/fun-emoji/svg?seed=Cleo", id:1},
-  {src:"https://api.dicebear.com/7.x/fun-emoji/svg?seed=Mittens", id:2},
-  {src:"https://api.dicebear.com/7.x/fun-emoji/svg?seed=Charlie", id:3},
-  {src:"https://api.dicebear.com/7.x/fun-emoji/svg?seed=Lucy", id:4},
-  {src:"https://api.dicebear.com/7.x/fun-emoji/svg?seed=Rascal", id:5},
-  {src:"https://api.dicebear.com/7.x/fun-emoji/svg?seed=Snowball", id:6},
-  {src:"https://api.dicebear.com/7.x/fun-emoji/svg?seed=Jack", id:7},
+  {src:sourceFront + "Ginger", id:0},
```
Comment (clairecassan) [inline src/App.tsx:8]: another way to write it: `${sourceFront}Ginger`

**Comment by syss69** [comment]
Done.


вт, 23 янв. 2024 г. в 11:17, clairecassan ***@***.***>:

File: src/App.tsx [hunk 13]
```diff
@@ -111,18 +113,24 @@ function App() {
...
+        if(sameImages){
+          increaseMatchCount();
+          setTimeout(() => {
+            setLastCardIndex(-1);
+            setPairClicks(1);
+          }, 500);
+        }
+        else{
+          setTimeout(() => {
+            if (!sameImages) {
```
Comment (clairecassan) [inline src/App.tsx:125]: testing sameImages is not necessary, you are already in a else block and it was tested in the if

**Comment by syss69** [comment]
Done


вт, 23 янв. 2024 г. в 13:11, clairecassan ***@***.***>:

**Pull request merged**